tests/integration/step-manager-test.js

```diff
@@ -56,10 +56,10 @@ module('step-manager', function(hooks) {
     });
 
     module('updating the target object from the component', function() {
-      test("mutates the target object's property when a mutable value is provided", async function(assert) {
+      test('onTransition callback is triggered with the current step when step changes', async function(assert) {
         this.set('step', 'first');
         await render(hbs`
-          {{#step-manager currentStep=(mut step) as |w|}}
+          {{#step-manager currentStep=this.step onTransition=(action (mut this.step)) as |w|}}
```

Comment:
This is fine, but you could also use `td` to create a stub function, something like

```js
this.onTransition = td.function();

...

assert.verify(this.onTransition('second'));
```

it's not a big difference and I wouldn't block the PR for it, but it allows you to avoid needing `mut` in the test and keeps the actual assertion in the test closer to what you're actually trying to verify.